gobblin-core/src/main/java/gobblin/qualitychecker/row/RowLevelErrFileWriter.java

```diff
@@ -32,17 +33,24 @@
  * @author stakiar
  */
 public class RowLevelErrFileWriter implements Closeable {
+  private final FileSystem fs;
   private final Closer closer = Closer.create();
   private BufferedWriter writer;
 
+  public RowLevelErrFileWriter(FileSystem fs) {
+    this.fs = fs;
+  }
+
   /**
    * Open a BufferedWriter
    * @param errFilePath path to write the file
    */
   public void open(Path errFilePath) throws IOException {
-    this.writer =
-        closer.register(new BufferedWriter(new OutputStreamWriter(new FileOutputStream(errFilePath.toString()),
-            ConfigurationKeys.DEFAULT_CHARSET_ENCODING)));
+    this.fs.mkdirs(errFilePath.getParent());
+    OutputStream os =
+        this.closer.register(this.fs.exists(errFilePath) ? this.fs.append(errFilePath) : this.fs.create(errFilePath));
```

Comment:
Since this is open source we can keep it. Append is not supposed to happen, since now `errFilePath` contains task state ID, so each task should write to a unique err file, unless someone uses the same row level policy twice in the same task.